log.go

```diff
@@ -57,3 +73,92 @@ func (l *Logger) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.Ha
 		"span":     time.Since(start).String(),
 	})
 }
+
+func GetSender(fn string) (send.Sender, error) {
+	const (
+		name        = "logkeeper"
+		interval    = 20 * time.Second
+		bufferCount = 100
+	)
+
+	var (
+		err       error
+		sender    send.Sender
+		senders   []send.Sender
+		baseLevel = send.LevelInfo{Default: level.Info, Threshold: level.Debug}
+	)
+
+	// configure remote logging services first if the environment
+	// variables are specified.
+
+	if splunk := send.GetSplunkConnectionInfo(); splunk.Populated() {
+		sender, err = send.NewSplunkLogger(name, splunk, baseLevel)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem creating the splunk logger")
+		}
+
+		senders = append(senders, send.NewBufferedSender(sender, interval, bufferCount))
+	}
+
+	if endpoint := os.Getenv("GRIP_SUMO_ENDPOINT"); endpoint != "" {
+		sender, err = send.NewSumo(name, endpoint)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem creating the sumo logic sender")
+		}
+		if err = sender.SetLevel(baseLevel); err != nil {
+			return nil, errors.Wrap(err, "problem setting level for alert remote object")
+		}
+
+		senders = append(senders, send.NewBufferedSender(sender, interval, bufferCount))
+	}
+
+	// configure slack logger for alerts and panics
+
+	if token := os.Getenv("GRIP_SLACK_CLIENT_TOKEN"); token != "" {
+		opts := &send.SlackOptions{
+			Name:          name,
+			Channel:       "#evergreen-ops-alerts",
+			AllFields:     true,
+			BasicMetadata: true,
+		}
+
+		sender, err = send.NewSlackLogger(opts, token, send.LevelInfo{Default: level.Error, Threshold: level.Critical})
+		if err != nil {
+			return nil, errors.Wrap(err, "problem creating the sumo logic sender")
+		}
+
+		// TODO use the amboy.Queue backed sender in this case.
+		senders = append(senders, sender)
+	}
+
+	// setup file logger, defaulting first to the system logger,
+	// to standard output (or not at all) if specified, and
+	// finally to the file as specified.
+
+	if fn == "" {
+		sender = getSystemLogger()
+		if err = sender.SetLevel(send.LevelInfo{Default: level.Info, Threshold: level.Debug}); err != nil {
+			return nil, errors.Wrap(err, "problem setting level for local system sender")
+		}
+
+		senders = append(senders, sender)
+	} else if fn == "NONE" || fn == "SKIP" {
```

Comment:
I would remove this and change line 154 to `else if fn != "NONE" || fn != "SKIP"`. It would seem to be more idiomatic and removes this "blank" block.

Reply:
awesome. will do, thanks!